BrowserKit/Tests/WebEngineTests/Mock/MockEngineSession.swift

```diff
@@ -0,0 +1,53 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import UIKit
+
+@testable import WebEngine
+
+class MockEngineSession: NSObject, EngineSession {
```

Comment:
It seems that some of the mocks are inheriting from a concrete class, @lmarceau do you have thoughts on why it was done that way and the benefits? In this case, I used the protocol method to keep the `PrintContentScript` tests isolated.

cc: @yoanarios

Reply:
`MockWKEngineSession` was one example. thanks!